app/pyproject.toml

```diff
@@ -30,7 +30,7 @@ pdfminer-six = "^20240706"
 unstructured = {extras = ["pdf", "docx", "pptx"], version = "^0.15.8"}
 nltk = "^3.9.1"
 [tool.poetry.group.dev.dependencies]
-black = "^23.9.1"
+black = "^24.8.0"
```

Comment:
Just curious what issue the 23.9.1 version was causing?

Reply:
Ah, I see the cannot parse issues. It might be more straightforward to break that up into two lines of code:

```
joined_headings = " > ".join(...)
context = f"some stuff... {joined_headings} ...other stuff"
```

than update black.